### PR TITLE
NoLoan with Gradle

### PR DESCRIPTION
**Pull request by gulilo**

### CODE REVIEW

File: projects/noloan2/app/src/main/java/hasadna/noloan2/FireStoreElement.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
+package hasadna.noloan2;
+
+
+//an element for storing proto type to Firestore
+public class FireStoreElement {
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/FireStoreElement.java]: s/FireStore/Firestore

File: projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+import javax.annotation.Nullable;
+
+public abstract class FirestoreAdapter<VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter implements EventListener<QuerySnapshot> {
+    private Query query;
+    private ListenerRegistration registration;
+
+    private ArrayList<DocumentSnapshot> snapshots = new ArrayList<>();
+
+    public FirestoreAdapter(Query query) {
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java]: Please use 2-space indentation.

File: projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+
+    @Override
+    public int getItemCount() {
+        return snapshots.size();
+    }
+
+    @Override
+    public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable FirebaseFirestoreException e) {
+        if (e != null) {
+            Log.w("firestore Adapter", e);
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java]: s/firestore Adapter/Firestore adapter

File: projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+        }
+    }
+
+    @Override
+    public int getItemCount() {
+        return snapshots.size();
+    }
+
+    @Override
+    public void onEvent(@Nullable QuerySnapshot queryDocumentSnapshots, @Nullable FirebaseFirestoreException e) {
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/FirestoreAdapter.java]: Perhaps:
s/queryDocumentSnapshots/querySnapshot

File: projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
+        setSupportActionBar(toolbar);
+
+        FloatingActionButton fab = (FloatingActionButton) findViewById(R.id.fab);
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java]: What does this `fab` do?

File: projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java [hunk 6]
```diff
@@ -0,0 +1,92 @@
...
+
+
+import noloan.R;
+
+public class MainActivity extends AppCompatActivity implements View.OnClickListener {
+
+    EditText name;
+    RecyclerView userView;
+
+    FirebaseFirestore mfirestore;
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java]: We don't use `m` prefixes. I suggest:
s/mfirestore/firestoreClient

File: projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java [hunk 7]
```diff
@@ -0,0 +1,92 @@
...
+        userView.setLayoutManager(new LinearLayoutManager(this));
+        adapter.startListening();
+    }
+
+    private void initFirestore() {
+        mfirestore = FirebaseFirestore.getInstance();
+        query = mfirestore.collection("Users");
+    }
+
+    //Write the user to the Firestore
```
Comment (oferb) [inline projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java]: Please add a space after the `//`. I.e from:
`//Write`
To:
`// Write`

Also in other places.

File: projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java (hunk 5 above)
Comment (gulilo) [reply projects/noloan2/app/src/main/java/hasadna/noloan2/MainActivity.java]: auto generated with the activity that i forgot to delete. gone now

**Pull request merged**